src/GafferSceneUI/ParameterInspector.cpp

```diff
@@ -108,11 +108,22 @@ Gaffer::ValuePlugPtr ParameterInspector::source( const GafferScene::SceneAlgo::H
 
 	if( auto light = runTimeCast<Light>( sceneNode ) )
 	{
-		if( auto optionalPlug = light->parametersPlug()->getChild<OptionalValuePlug>( m_parameter.name ) )
+		if( m_parameter.shader.string().empty() )
 		{
-			return optionalPlug->enabledPlug()->getValue() ? optionalPlug : nullptr;
+			if( auto optionalPlug = light->parametersPlug()->getChild<OptionalValuePlug>( m_parameter.name ) )
+			{
+				return optionalPlug->enabledPlug()->getValue() ? optionalPlug : nullptr;
+			}
+		}
+		/// \todo Remove the need to search for a `ShaderPlug` by adding such a plug to
+		/// `GafferScene::Light` itself.
+		for( const auto &plug : Plug::Range( *light ) )
+		{
+			if( const auto shaderPlug = runTimeCast<ShaderPlug>( plug ) )
+			{
+				return const_cast<ValuePlug *>( shaderPlug->parameterSource( m_parameter ) );
```

Comment:
We can use `parameterSource()` in the ShaderAssignment case below as well - it's even easier there, and it means we can drop the `Switch` handling shenanigans because that is dealt with for us in `parameterSource()` now.